homeassistant/components/powerwall/config_flow.py

```diff
@@ -0,0 +1,76 @@
+"""Config flow for Tesla Powerwall integration."""
+import logging
+
+from tesla_powerwall import ApiError, PowerWall, PowerWallUnreachableError
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_IP_ADDRESS
+
+from .const import DOMAIN  # pylint:disable=unused-import
+from .const import POWERWALL_SITE_NAME
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_IP_ADDRESS): str})
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+
+    power_wall = PowerWall(data[CONF_IP_ADDRESS])
+
+    def _call_site_info(power_wall):
```

Comment:
Can we reuse the function we defined in the component module?

Reply:
yes done